docs/ml-process.md

```diff
@@ -0,0 +1,318 @@
+
+## Prerequisites
+
+This section is required if you want to make changes to Bugbug codebase (any of the first 3 steps).
+
+You will need to create a fork of https://github.com/mozilla/bugbug
+
+Then install required dependencies as described in https://github.com/mozilla/bugbug#setup-and-prerequisites:
+
+```
+$ git clone https://github.com/<your github username>/bugbug
+$ cd bugbug
+$ python3 -m venv env
+$ . env/bin/activate
+$ pip3 install -r requirements.txt
+```
+
+Note that Bugbug http service has its own dependencies, and instructions on how
+to run it are located here: https://github.com/mozilla/bugbug/blob/master/http_service/README.md#local-development
+
+## 1. Data collection
+
+During this step we retrieve all issues via GitHub API from [webcompat web-bugs repository](https://github.com/webcompat/web-bugs/issues) and 
+store it in a file with [JSON Lines](https://jsonlines.org/) format. This file is our dataset and it will be used for model training in [step 2](#training-the-model).
+It's also worth mentioning that this json file is compressed using [Zstandard](https://facebook.github.io/zstd/).
+
+The script responsible for data fetching is [located here](https://github.com/mozilla/bugbug/blob/master/scripts/github_issue_retriever.py).
+If you wish to run it locally and can be done with the following command:
+
+```
+$ python3 -m scripts.github_issue_retriever --owner=webcompat --repo=web-bugs --retrieve-events
+```
+
+`--retrieve-events` in this command will make sure that each issue events data is retrieved and stored in the dataset. 
+Issue events are triggered by activity in issues (renaming, change of milestone, labelling, etc) and describing the lifecycle of an issue.
+This parameter is required as events are used for our model training (more on that below).
+
+Issues retrieval task is scheduled to run once in two weeks in the Taskcluster, together with the model training task. 
+The task that is running this script [can be found here](https://github.com/mozilla/bugbug/blob/3e4f452b5e523d3d114cb0477e5a82f22c7e951e/infra/data-pipeline.yml#L232-L270) 
+and results of the task are [located here](https://community-tc.services.mozilla.com/tasks/index/project.bugbug.data_github_issues/latest).
+
+If you run the above command locally, it will look for the existing dataset and check its `last_modified` timestamp (the dataset is saved as an [artifact of the task](https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_github_issues.latest/artifacts/public%2Fgithub_issues.json.zst) in the Taskcluster). 
+If the dataset is available, it will be downloaded to `bugbug/data` folder to your machine and the script will begin fetching updates from GitHub and save them to this local copy. 
+
+During this update, all changes to issues since `last_modified` timestamp will be fetched. If the dataset or its `last_modified` is not available, the script will begin fetching all issues starting from issue 1.
+
```

Comment:
It could be interesting to understand how long it might take to download all issues. It takes xxx minutes for 10,000 issues.

Reply:
I've added more on this below + GitHub rate limiting rules